### PR TITLE
Add support for Gradle v7, fixes #164

### LINKED ISSUE #164: Add support for Gradle v7
Following up from https://github.com/sourcegraph/lsif-java/pull/163

The CI started failing recently after the Gradle v7 release. See the PR for a full stack trace.

### PR DESCRIPTION
**Pull request by olafurpg**
Also adds a custom error message for users on Gradle v6.7 using the
experimental toolchains feature. We can only support toolchains from
version v6.8.

### CODE REVIEW

File: lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleJavaToolchains.scala [hunk 1]
```diff
@@ -53,9 +55,20 @@ object GradleJavaToolchains {
...
     val toolchainsPath = tmp.resolve("java-toolchains.txt")
+    val gradleVersionPath = tmp.resolve("gradle-version.txt")
     val taskName = "lsifDetectJavaToolchains"
     val script =
-      s"""|allprojects {
+      s"""|
+          |try {
+          |  java.nio.file.Files.write(
+          |    java.nio.file.Paths.get('$gradleVersionPath'),
+          |    [gradle.gradleVersion],
```
Comment (Strum355) [inline lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleJavaToolchains.scala:65]: Is this some scala feature Im not aware of?

File: lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala [hunk 2]
```diff
@@ -37,31 +37,49 @@ class GradleBuildTool(index: IndexCommand) extends BuildTool("Gradle", index) {
...
+              toolchains.toolchains.nonEmpty =>
+          index
+            .app
+            .error(
+              "lsif-java does not support Gradle 6.7 when used together with Java toolchains. " +
+                "To fix this problem, upgrade to Gradle version 6.8 or newer and try again."
+            )
+          CommandResult(1, Nil)
+        case _ =>
+          runCompileCommand(toolchains)
```
Comment (Strum355) [inline lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala:55]: Will this still call `runCompileCommand` if `if gradleVersion.startsWith("6.7") && toolchains.toolchains.nonEmpty` evaluates to false?

**Review by Strum355 — APPROVED** [review]

File: lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala (hunk 2 above)
Comment (olafurpg) [reply lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala:55]: Yes, `case _ =>` is a catchall

File: lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleJavaToolchains.scala (hunk 1 above)
Comment (olafurpg) [reply lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleJavaToolchains.scala:65]: Which feature? This is a Groovy script, where you have literal syntax `[a]` for lists.

**Pull request merged**